src/db_type/output.rs

```diff
@@ -15,12 +15,13 @@ impl DatabaseOutputValue {
     }
 }
 
-pub(crate) fn unwrap_item<T: Input>(item: Option<redb::AccessGuard<&'static [u8]>>) -> Option<T> {
-    if let Some(item) = item {
-        let item = item.value();
-        let item = T::native_db_bincode_decode_from_slice(item);
-        Some(item)
-    } else {
-        None
-    }
```

Comment:
> It's less idiomatic, but it is also personal preference. I personally find it more readable with the map, as it makes it more explicit, that we operate on self.
> It's less idiomatic, I would say. As essentially you are destructing an Option to then return an Option. It would be different, if we would return item. If you are working with Option and returning Option, I find it more idiomatic, work within the option.


Maybe you are making me change my mind. I understand your arguments are relevant. I just have a problem with the aesthetics/visibility/symmetry of the code...